### PR TITLE
fix: refactor bunch of wallets to TS

### PR DESCRIPTION
**Pull request by limpbrains**
Nothing fancy. Mostly copy-pasting existing styles and adding throws to protect types

### CODE REVIEW

File: class/wallets/hd-legacy-p2pkh-wallet.ts [hunk 1]
```diff
@@ -71,12 +76,19 @@ export class HDLegacyP2PKHWallet extends AbstractHDElectrumWallet {
       newUtxos.push(u);
     }
 
-    return newUtxos;
```
Comment (limpbrains) [inline class/wallets/hd-legacy-p2pkh-wallet.ts:74]: notable change

File: class/wallets/hd-segwit-p2sh-wallet.ts [hunk 2]
```diff
@@ -90,7 +105,7 @@ export class HDSegwitP2SHWallet extends AbstractHDElectrumWallet {
       ],
       witnessUtxo: {
         script: p2sh.output,
-        value: input.amount || input.value,
```
Comment (limpbrains) [inline class/wallets/hd-segwit-p2sh-wallet.ts:93]: notable change
Is ti safe ?

File: class/wallets/lightning-ldk-wallet.ts [hunk 3]
```diff
@@ -564,6 +564,7 @@ export class LightningLdkWallet extends LightningCustodianWallet {
     await wallet.fetchUtxo();
     console.log(wallet.getBalance(), wallet.getUtxo());
     console.log('creating transation...');
+    // @ts-ignore wtf wallet.getUtxo() and first arg of createTransaction are not compatible
```
Comment (limpbrains) [inline class/wallets/lightning-ldk-wallet.ts:567]: wtf

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/orswgetlcpagrv4ftlyil3g5ke?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/488)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/zuirq7gvpyi5zfkmvptpzwlxem?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/496)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/laelsf2z5hjxezlps4bgeedzbq?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/506)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/wgwo73k2a57bzv5u3csyhwelxe?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/543)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/e26ipxk57tygg3yujbgs3ucgs4?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/544)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/swvr36a4pgmbr63x63nln7srwu?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/545)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/esbzfna4cegfsibbuqtwofyhqu?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/551)

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @marcosrdz] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/marcosrdz

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @marcosrdz] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/marcosrdz

**Comment by Overtorment** [comment]
i have just merged some tests to master. i want them to pass on this PR before i look at it again

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/6t4ue5lqc7id4y4pngo2l365yy?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/588)

File: class/wallets/hd-segwit-p2sh-wallet.ts (hunk 2 above)
Comment (Overtorment) [reply class/wallets/hd-segwit-p2sh-wallet.ts:93]: i checked the code, should be safe. 
it always boils down to `fetchUtxo`, then to  `blockchainScripthash_listunspentBatch` which has `value`

https://electrumx.readthedocs.io/en/latest/protocol-methods.html#blockchain-scripthash-listunspent

File: class/wallets/legacy-wallet.ts [hunk 4]
```diff
@@ -430,11 +426,12 @@ export class LegacyWallet extends AbstractWallet {
     const values: Record<number, number> = {};
     let keyPair: Signer | null = null;
 
+    if (!skipSigning) {
+      // skiping signing related stuff
+      keyPair = ECPair.fromWIF(this.secret); // secret is WIF
```
Comment (Overtorment) [inline class/wallets/legacy-wallet.ts:431]: nice optimization

File: class/wallets/lightning-ldk-wallet.ts (hunk 3 above)
Comment (Overtorment) [reply class/wallets/lightning-ldk-wallet.ts:567]: a bit weird, yeah. we should probably unify  types `Utxo` &  `CreateTransactionUtxo` , then this incompatibility will go away

**Review by Overtorment — APPROVED** [review]

**Pull request merged**